tests/sparseml/optim/test_helpers.py

```diff
@@ -117,21 +163,29 @@ def _test_nested_equality(val, other):
 
 
 @pytest.mark.parametrize(
-    "recipe,expected_recipe",
+    "recipe,expected_recipe, is_staged",
     [
-        (TARGET_RECIPE.format(num_epochs=10.0), TARGET_RECIPE.format(num_epochs=10.0)),
-        (RECIPE_SIMPLE_EVAL, TARGET_RECIPE.format(num_epochs=10.0)),
-        (RECIPE_MULTI_EVAL, TARGET_RECIPE.format(num_epochs=10.0)),
+        (
+            TARGET_RECIPE.format(num_epochs=10.0),
+            TARGET_RECIPE.format(num_epochs=10.0),
+            False,
+        ),
+        (RECIPE_SIMPLE_EVAL, TARGET_RECIPE.format(num_epochs=10.0), False),
+        (RECIPE_MULTI_EVAL, TARGET_RECIPE.format(num_epochs=10.0), False),
+        (STAGED_RECIPE, STAGED_RECIPE, True),
```

Comment:
the given and expected recipe should not be the same, let's make sure that we get the expected recipe added and verify that the proper checks are made so the test only passes when the eval is correct